2020/dec12/ship/def.go

```diff
@@ -0,0 +1,24 @@
+package ship
+
+type Function int
+
+const (
+	North Function = iota
+	South
+	East
+	West
+	Port
+	Star
+	Forward
+)
+
+type Instruction struct {
+	Function Function
+	Value    int
+}
+
+type Runner interface {
+	Run() error
+	Latitude() int
+	Longitude() int
```

Comment:
I'd go with X() / Y(), or North(), East() (or South(), West(), up to you), to avoid magellanic shenanigans.